src/lib/tableland-calls.ts

```diff
@@ -1,55 +1,44 @@
 /* eslint-disable node/no-missing-import */
 import camelCase from "camelcase";
 import {
-  TableMetadata,
   StructureHashReceipt,
   RpcParams,
-  RpcRequestParam,
   ReadQueryResult,
+  WriteQueryResult,
   ReceiptResult,
   KeyVal,
   Connection,
 } from "../interfaces.js";
 import { list } from "./list.js";
 
 async function SendCall(this: Connection, rpcBody: Object) {
-  return await fetch(`${this.host}/rpc`, {
+  const res = await fetch(`${this.host}/rpc`, {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
       Authorization: `Bearer ${this.token.token}`,
     },
     body: JSON.stringify(rpcBody),
   });
+
+  return parseResponse(res);
 }
 
 // parse the rpc response and throw if any of the different types of errors occur
```

Comment:
I refactored this so that the different structures returned by the RPC calls are handled in the method's body.  This way we don't have to do stuff like `return json.some.key as SomeType`

Reply:
Yeh nice.